tests/test_arrow_dataset.py

```diff
@@ -3618,6 +3620,67 @@ def test_dataset_from_generator_features(features, data_generator, tmp_path):
     _check_generator_dataset(dataset, expected_features)
 
 
+def _create_spark_dataframe(tmp_dir):
```

Comment:
Could we split the test to a different class so that we can put the spark creation into `setUp` function?

Reply:
I reverted this change